File: api/get_ticket_chatter.py
from fastapi import FastAPI, Request
from utils.odoo import get_ticket_chatter
import os

app = FastAPI()

@app.get("/api/get_ticket_chatter")
async def handler(request: Request):
    api_key = request.headers.get("x-api-key")
    if api_key != os.environ.get("MCP_API_KEY"):
        return {"error": "Invalid API Key"}

    ticket_id = request.query_params.get("ticket_id")
    if not ticket_id:
        return {"error": "Missing ticket_id"}

    try:
        ticket_id = int(ticket_id)
        messages = get_ticket_chatter(ticket_id)
        return {"ticket_id": ticket_id, "count": len(messages), "messages": messages}
    except Exception as e:
        return {"error": str(e)}

File: api/get_ticket_chatter.ts
import type { VercelRequest, VercelResponse } from "@vercel/node";
import { getTicketChatter } from "../utils/odoo.js";

export default async function handler(req: VercelRequest, res: VercelResponse) {
  const apiKey = req.headers["x-api-key"];
  if (apiKey !== process.env.MCP_API_KEY) {
    return res.status(401).json({ error: "Invalid API Key" });
  }

  const ticket_id = parseInt(req.query.ticket_id as string);

  if (!ticket_id) {
    return res.status(400).json({ error: "Missing ticket_id" });
  }

  try {
    const messages = await getTicketChatter(ticket_id);
    return res.status(200).json({
      ticket_id,
      count: messages.length,
      messages
    });
  } catch (err: any) {
    return res.status(500).json({ error: err.message });
  }
}